cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -51,33 +53,37 @@ public SchemaService(ClusterApiUtils clusterApiUtils) {
   }
 
   public synchronized ApiResponse registerSchema(ClusterSchemaRequest clusterSchemaRequest) {
-    String schemaCompatibility = null;
+    Pair<String, String> schemaCompatibility = null;
     boolean schemaCompatibilityCompleted = false;
     try {
       log.info("RegisterSchema on {} ", clusterSchemaRequest.getTopicName());
       //            set default compatibility
       if (clusterSchemaRequest.isForceRegister()) {
         log.debug("RegisterSchema - Force Register Enabled");
         schemaCompatibility =
-            getSchemaCompatibility(
+            getSubjectCompatibilityWithGlobalFallBack(
                 clusterSchemaRequest.getEnv(),
                 clusterSchemaRequest.getTopicName(),
                 clusterSchemaRequest.getProtocol(),
                 clusterSchemaRequest.getClusterIdentification());
         // Error thrown preventing progress if Schema Compatibility not retrieved correctly.
         schemaCompatibilityCompleted =
-            checkSchemaCompatibilitySuccessfullyRetrieved(schemaCompatibility);
+            checkSchemaCompatibilitySuccessfullyRetrieved(schemaCompatibility.getLeft());
         log.debug(
             "RegisterSchema - original Schema Compatibility {} for Topic {}",
             schemaCompatibility,
             clusterSchemaRequest.getTopicName());
-        setSchemaCompatibility(
-            clusterSchemaRequest.getEnv(),
-            clusterSchemaRequest.getTopicName(),
-            clusterSchemaRequest.isForceRegister(),
-            clusterSchemaRequest.getProtocol(),
-            clusterSchemaRequest.getClusterIdentification(),
-            null);
+        // only reset if we have the previous schema to revert to.
+        if (schemaCompatibilityCompleted) {
```

Comment:
If the schemaCompatibility is coming from global, we should set the compatibility to NONE even. So we can skip this call in that case